tests/tests-memcached/src/main/java/com/hazelcast/simulator/memcached/MemcachedClientFactory.java

```diff
@@ -0,0 +1,65 @@
+package com.hazelcast.simulator.memcached;
+
+import net.spy.memcached.MemcachedClient;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MemcachedClientFactory {
+
+    protected final Logger logger = Logger.getLogger(getClass());
+
+    private List<MemcachedClient> clients = new ArrayList<MemcachedClient>();
+    private List<InetSocketAddress> addresses;
+    private boolean useSharedClient;
+
+    public MemcachedClientFactory(String[] nodes) {
+        List<InetSocketAddress> addresses = new ArrayList<InetSocketAddress>();
+        for (String node: nodes) {
+            String[] addressParts = node.split(":");
+            if (addressParts.length == 0 || addressParts.length > 2) {
+                throw new IllegalArgumentException("Invalid node address. Example: localhost:11211");
+            }
+
+            int port = 11211; //default memcached port
+            if (addressParts.length == 2) {
+                port = Integer.parseInt(addressParts[1]);
+            }
+            addresses.add(new InetSocketAddress(addressParts[0], port));
+        }
+
+        this.addresses = addresses;
+    }
+
+    public MemcachedClient create() {
+        if (useSharedClient && clients.size() == 1) {
+            logger.info("Using a shared client");
+            return clients.get(0);
+        }
+
+        logger.info("Creating a new client");
+
+        MemcachedClient client = null;
```

Comment:
this can lead to more than 1 client being created. Keep things simple and just make the method synchronized